### PR TITLE
Change version generation

### PR DESCRIPTION
**Pull request by babich-a**
Default flavors were added:
'beta' for labels
'alpha', 'build' for builds
'dev' - as is

Hours/minutes tail was added.

It can be tested with the following script. Take in mind that DEVEXTREME_DXBUILD_LABEL  and DEVEXTREME_DXBUILD_REVISION  can't be set at the same time.
```
// process.env.DEVEXTREME_DXBUILD_LABEL = '19_1_0';
// process.env.DEVEXTREME_DXBUILD_FLAVOR = 'beta2';
// process.env.DEVEXTREME_DXBUILD_REVISION = '18091';

console.log(require('./context'));
```

### CODE REVIEW

**Comment by AlekseyMartynov** [comment]
🤞 and good luck.
A unit test would be great to describe the expected versions.

**Comment by babich-a** [comment]
> 
> 
> 🤞 and good luck.
> A unit test would be great to describe the expected versions.

@AlekseyMartynov,  I've added the testing file.

**Comment by AlekseyMartynov** [comment]
@San4es
Please check version-spec.js
Does it cover all combinations that we need?

File: build/gulp/version.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        : version;
+};
+
+const getRevision = (revision) => {
+    if(!revision) return '';
+
+    const date = new Date();
+    const padStart = (number) => ('0' + number).slice(-2);
+
+    const hours = padStart(date.getHours());
```
Comment (San4es) [inline build/gulp/version.js:31]: How do we plan to sync this tail with NuGet packages? DEVEXTREME_DXBUILD_TIMESTAMP? DEVEXTREME_DXBUILD_REVISION?

Comment (babich-a) [reply build/gulp/version.js:31]: We'll get the version from  the `package.json` of npm package for bower and nuget

**Review by San4es — APPROVED** [review]
LGTM

**Pull request merged**